src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2213,7 +2213,19 @@ void RadioSpinel<InterfaceType, ProcessContextType>::CalcRcpTimeOffset(void)
 template <typename InterfaceType, typename ProcessContextType>
 uint64_t RadioSpinel<InterfaceType, ProcessContextType>::GetNow(void)
 {
-    return mIsTimeSynced ? (otPlatTimeGet() + static_cast<uint64_t>(mRadioTimeOffset)) : UINT64_MAX;
+    uint64_t now = UINT64_MAX;
+    if (mIsTimeSynced)
```

Comment:
Thanks @thcu-gp .

Can we instead define `mRadioTimeOffset` as `uint64` directly (instead of `int64_t`).

We calculate it as
``` 
mRadioTimeOffset  = (remoteTimestamp - ((localRxTimestamp / 2) + (localTxTimestamp / 2)));
```
where everything is `uint64_t` already and later we use

```
otPlatTimeGet() + mRadioTimeOffset
```

I think the calculations would be correct even if the result of subtractions is effectively negative (as long as all the calculations are using `uint64_t`).

Reply:
I can give it a whirl (probably somewhere next week)